src/repositories/expenseRepository.js

```diff
@@ -0,0 +1,95 @@
+const mockExpenses = require('../api/mockExpenses');
+
+const expenseRepository = {
+  create: (expenseData) => {
+    const newExpense = {
+      id: mockExpenses.length + 1,
+      ...expenseData,
+    };
+
+    mockExpenses.push(newExpense);
+
+    return newExpense;
+  },
+
+  findAll: (filters = {}) => {
+    let expenses = mockExpenses;
+
+    const { userId, from, to, categories } = filters;
+
+    if (userId) {
+      expenses = expenses.filter(
+        (expense) => expense.userId === Number(userId),
+      );
+    }
+
+    if (from) {
+      const fromDate = new Date(from);
+
+      expenses = expenses.filter(
+        (expense) => new Date(expense.spentAt) >= fromDate,
+      );
+    }
+
+    if (to) {
+      const toDate = new Date(to);
+
+      expenses = expenses.filter(
+        (expense) => new Date(expense.spentAt) <= toDate,
+      );
+    }
+
+    if (categories && categories.length > 0) {
+      /* eslint-disable */
+      const normalizedCategories = Array.isArray(categories)
+        ? categories.map((cat) => cat.toLowerCase())
+        : categories.split(',').map((cat) => cat.trim().toLowerCase());
+
+      if (normalizedCategories.length > 0) {
+        expenses = expenses.filter((expense) =>
+          normalizedCategories.includes(expense.category.toLowerCase()),
+        );
+      }
+      /* eslint-enable */
+    }
+
```

Comment:
All these if approaches work, but could become something hard to maintain and scale in case of more and more properties/queries needed.
You could try storing filtering functions in a hashMap (Object works), and using the given function accordingly.
This will keep all scalability in a single place and remove multiple if checks. (because your code checks all ifs, even when already passing the first one).
If you do so, remember to add a default filtering function